modules/trivy-operator/main.tf

```diff
@@ -17,27 +17,22 @@ resource "helm_release" "trivy-operator" {
   values = [templatefile("${path.module}/templates/values-trivy-operator.yaml", {})]
 }
 
-resource "kubernetes_manifest" "vmservicescrape" {
-  manifest = {
-    apiVersion = "operator.victoriametrics.com/v1beta1"
-    kind       = "VMServiceScrape"
-    metadata = {
-      name      = "trivy-vmservicescrape"
-      namespace = kubernetes_namespace.trivy-system.metadata[0].name
-    }
-    spec = {
-      endpoints = [
-        {
-          port = "metrics"
-        }
-      ]
-      selector = {
-        matchLabels = {
-          "app.kubernetes.io/name" = "trivy-operator"
-        }
-      }
-    }
-  }
+resource "kubectl_manifest" "vmservicescrape" {

```

Comment:
Swapping over to this is to resolve a race condition issue. Since the VMServiceScrape kind is a resource that depends on a CRD (Custom Resource Definition) being installed already on the cluster the terraform plan stage was failing with:

```
│ Error: API did not recognize GroupVersionKind from manifest (CRD may not be installed)
│ 
│   with module.trivy-operator.kubernetes_manifest.vmservicescrape,
│   on .terraform/modules/trivy-operator/main.tf line 20, in resource "kubernetes_manifest" "vmservicescrape":
│   20: resource "kubernetes_manifest" "vmservicescrape" {
│ 
│ no matches for kind "VMServiceScrape" in group
│ "operator.victoriametrics.com"
```

The suggestion was to try out a different provider (https://registry.terraform.io/providers/gavinbunney/kubectl/latest/docs/resources/kubectl_manifest) to get around this issue. This allows the resources to be created as expected in the right order.

Reply:
Interesting that thisre resource has you pass a whole YAML configuration to the `yaml_body` arg instead of somthing more similar to how `kubernetes_manifest` is configured